wsmaster/che-core-api-factory-bitbucket-server/src/main/java/org/eclipse/che/api/factory/server/bitbucket/BitbucketServerPersonalAccessTokenFetcher.java

```diff
@@ -97,4 +102,25 @@ public PersonalAccessToken fetchPersonalAccessToken(Subject cheUser, String scmS
       throw new ScmCommunicationException(e.getMessage(), e);
     }
   }
+
+  @Override
+  public Optional<Boolean> isValid(PersonalAccessToken personalAccessToken)
+      throws ScmCommunicationException, ScmUnauthorizedException {
+    if (!bitbucketServerApiClient.isConnected(personalAccessToken.getScmProviderUrl())) {
+      LOG.debug(
+          "not a  valid url {} for current fetcher ", personalAccessToken.getScmProviderUrl());
+      return Optional.empty();
+    }
+    try {
+      BitbucketPersonalAccessToken bitbucketPersonalAccessToken =
+          bitbucketServerApiClient.getPersonalAccessToken(
+              personalAccessToken.getScmUserName(),
+              Long.valueOf(personalAccessToken.getScmTokenId()));
+      return Optional.of(
+          Sets.difference(DEFAULT_TOKEN_SCOPE, bitbucketPersonalAccessToken.getPermissions())
```

Comment:
`DEFAULT_TOKEN_SCOPE.equals(bitbucketPersonalAccessToken.getPermissions())` will do the same job, is more explicit and without need to use 3rd party library